app/Services/MovieServiceImpl.php

```diff
@@ -51,13 +55,46 @@ public function count(){
         return Movie::count();
     }
     public function create($movie){
+        $image = $movie['image'];
+        $savedImage = null;
+        if ($image){
+            $image_name = time() . '.' . $image->getClientOriginalExtension();
+
+            $destinationFullSize = public_path('storage/fullSize').'/'.$image_name;
+            $destinationThumbnail = public_path('storage/thumbnail').'/'.$image_name;
+
+            $createdImage= Image::make($image->getRealPath());
+
+            $createdImage->resize(400, 400, function($constraint){
+            $constraint->aspectRatio();
+            })->save($destinationFullSize);
+
+            $createdImage->resize(200, 200, function($constraint){
+                $constraint->aspectRatio();
+            })->save($destinationThumbnail);
+
+            $savedImage = MovieImage::create([
+                'fullSize' => asset('storage/fullSize/'.$image_name),
+                'thumbnail' => asset('storage/thumbnail/'.$image_name)
+
+            ]);
+        }else if ($movie['image_url']){ //this is used when creating movie from OMDB
+            $savedImage = MovieImage::create([
+                'fullSize' => $movie['image_url'],
+                'thumbnail' => $movie['image_url'],
+
+            ]);
+        }else{
+            abort(422, "Image not presented.");
+        }
         return Movie::create([
             'title' => $movie['title'],
             'description' => $movie['description'],
-            'image_url' => $movie['image_url'],
             'num_of_visits' => 0,
             'genre_id' => $movie['genre_id'],
+            'image_id' => $savedImage->id,
```

Comment:
ovako snimaj relacije https://laravel.com/docs/6.x/eloquent-relationships#the-save-method